riva/proto/riva_nmt.proto

```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
   // Multiple texts may be passed per request up to the given batch size for the model, which is set at translation pipeline creation time.
-rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
+  rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
 
   // Lists the available language pairs and models names to be used for TranslateText
   rpc ListSupportedLanguagePairs(AvailableLanguageRequest) returns (AvailableLanguageResponse) {}
+
+  //streaming speech to text translation api.
+  rpc StreamingTranslateSpeechToText(stream StreamingTranslateSpeechToTextRequest)
+        returns (stream StreamingTranslateSpeechToTextResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToText(TranslateSpeechToTextRequest) returns (TranslateSpeechToTextResponse) {}
+
+  //Returns the parameters for the configurated pipelines.
+  //rpc GetTranslateSpeechToTextConfig(GetTranslateSpeechToTextConfigRequest) returns (GetTranslateSpeechToTextConfigResponse) {}
+
+  rpc StreamingTranslateSpeechToSpeech(stream StreamingTranslateSpeechToSpeechRequest)
+    returns (stream StreamingTranslateSpeechToSpeechResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToSpeech(TranslateSpeechToSpeechRequest) returns (TranslateSpeechToSpeechResponse) {}
+
+
 }
 
+/*
+* Configuration for Translate S2S.  reuse existing protos from other services.
+*/
+message StreamingTranslateSpeechToSpeechConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig asrConfig = 1;  //from riva_asr.proto
+   SynthesizeSpeechConfig ttsConfig = 2; //from riva_tts.proto
+   string source_language_code = 3;
+   string target_language_code = 4;
+   AudioEncoding t = 5;
+}
+
+/*
+* Streaming translate speech to speech used to configure the entire pipline for speech translation.  This can be
+* be backed by a cascade of ASR, NMT, TTS models or an end to end model
+*
+*/
+message StreamingTranslateSpeechToSpeechRequest {
+   oneof streaming_request {
+       StreamingTranslateSpeechToSpeechConfig config = 1;
+       bytes audio_content = 2;
+   }
+}
+
+message SynthesizeSpeechConfig {
+        AudioEncoding encoding = 1;
+        int32 sample_rate_hz = 2;
+        string voice_name = 3;
+}
+
+/*
+*
+*/
+message StreamingTranslateSpeechToSpeechResponse {
+        nvidia.riva.tts.SynthesizeSpeechResponse speech = 1; //from riva_tts.proto
+}
+
+message StreamingTranslateSpeechToTextRequest {
+  oneof streaming_request {
+     StreamingTranslateSpeechToTextConfig config = 1;
+     bytes audio_content = 2;
+  }
+}
+
+message StreamingTranslateSpeechToTextResponse {
+
+   repeated nvidia.riva.asr.StreamingRecognitionResult results = 1; //from riva_asr.proto
+}
+
+message StreamingTranslateSpeechToTextConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig config = 1; //existing ASR config
+   bool interim_results = 2;  // not always supported
```

Comment:
Is this different than `interim_results` in asr streaming config?

Reply:
this is intended to allow the interim ASR transcription be returned alongside the translated text, will remove for now